### PR TITLE
docs: Add separate information about scripts

### PR DESCRIPTION
**Pull request by tgodzik**
We added some info in the previous release notes, but it's better to keep it in stable documentation.

I think we can start moving some feature docs to features tab to be easier to discover.

### CODE REVIEW

File: docs/features/scripts.md [hunk 1]
```diff
@@ -0,0 +1,50 @@
+---
+id: scripts
+title: Scripts support
+---
+
+### Working with a script
+
+Whenever, Metals opens a script file with `*.sc` extension, but only when it's
+not `*.worksheet.sc` or a build server is not already started for it, users will
```
Comment (LaurenceWarne) [inline docs/features/scripts.md:9]:
```suggestion
not `*.worksheet.sc` or a build server has not already started for it, users will
```

File: docs/features/scripts.md [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+be prompted to choose whether they want to use [Ammonite](http://ammonite.io/)
+or [ScalaCli](https://scala-cli.virtuslab.org/) for using within the it.
```
Comment (LaurenceWarne) [inline docs/features/scripts.md:11]:
```suggestion
or [ScalaCli](https://scala-cli.virtuslab.org/) for usage within it.
```

Comment (ckipp01) [inline docs/features/scripts.md:11]: Just a nit, but I never see `ScalaCLI` all as one word like this. Would it be more accurate to say `Scala CLI`?

Comment (ckipp01) [reply docs/features/scripts.md:11]: I actually think this is worded a bit oddly. I'l recommend actually changing the whole ending to something like:

> for it.

If you want to keep it really simple, but if you want to be more detailed something like:

> the power the script.

File: docs/features/scripts.md [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+![scala-cli](https://i.imgur.com/ghR1Src.gif)
+
+Scripts are usually used for smaller programs and the main difference is that
+statements can be put on top level without a need for additional objects or
+methods.
+
+Once user have chosen a specific scripting tool, Metals will also suggest to
```
Comment (ckipp01) [inline docs/features/scripts.md:19]:
```suggestion
Once a user has chosen a specific scripting tool, Metals will also suggest to
```

File: docs/features/scripts.md [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+import all newly opened scripts automatically or to always to do that manually.
+
+Both tools will start a build server in the background that will be able to
+compile your code and provide Metals with all the necessary information.
+
+ScalaCLI scripts can also be used standalone when ScalaCli generated `.bsp`
```
Comment (ckipp01) [inline docs/features/scripts.md:25]: I see both `ScalaCLI` and `ScalaCli`. Whatever you choose you should use the same throughout.

File: docs/features/scripts.md (hunk 2 above)
Comment (tgodzik) [reply docs/features/scripts.md:11]: Actually it's `Scala CLI` based on what we wrote on the website :sweat_smile:

Comment (tgodzik) [reply docs/features/scripts.md]: Changed everywhere I think!

File: docs/features/scripts.md (hunk 3 above)
Comment (tgodzik) [reply docs/features/scripts.md]: done!

**Comment by tgodzik** [comment]
> Nice! Suggestion: add a link to/something along thelines of https://scalameta.org/metals/docs/troubleshooting/faq/#how-do-i-use-scala-2xx-for-my-script, I think it can sometimes be easy to lookover it there: [emacs-lsp/lsp-metals#78](https://github.com/emacs-lsp/lsp-metals/issues/78)

Added a link to the FAQ at the bottom.

File: docs/features/scripts.md [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+statements can be put on top level without a need for additional objects or
+methods.
+
+Once a user has chosen a specific scripting tool, Metals will also suggest to
+import all newly opened scripts automatically or to always to do that manually.
+
+Both tools will start a build server in the background that will be able to
+compile your code and provide Metals with all the necessary information.
+
+ScalaCLI scripts can also be used standalone when Scala CLI generated `.bsp`
```
Comment (ckipp01) [inline docs/features/scripts.md:25]:
```suggestion
Scala CLI scripts can also be used standalone when Scala CLI generated `.bsp`
```

File: docs/features/scripts.md [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+and scala files in that directory as ones that can be used together. So you
+would be able to import method and classes from those files. However, if the
+script is contained within a directory that also contains other sources, that
+script will be treated as a standalone one in order to avoid flaky compilation
+errors coming from normal files in the workspace.
+
+Current limitations can be found:
+
+- [here for Ammonite](https://github.com/scalameta/metals/issues?q=is%3Aopen+is%3Aissue+label%3A%22ammonite+support%22)
+- [here for ScalaCLI](https://github.com/scalameta/metals/issues?q=is%3Aopen+is%3Aissue+label%3Ascala-cli)
```
Comment (ckipp01) [inline docs/features/scripts.md:50]:
```suggestion
- [here for Scala CLI](https://github.com/scalameta/metals/issues?q=is%3Aopen+is%3Aissue+label%3Ascala-cli)
```

**Review by ckipp01 — APPROVED** [review]
Ha just a couple more ScalaCLI -> Scala CLI nits, but apart from that LGTM!

**Comment by tgodzik** [comment]
> Ha just a couple more ScalaCLI -> Scala CLI nits, but apart from that LGTM!

My eys were not successful in scanning the names :sweat_smile:

**Pull request merged**